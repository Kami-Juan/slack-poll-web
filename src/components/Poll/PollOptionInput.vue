<template>
  <b-row align-h="center" id="poll-option-input">
    <b-col cols=6>
      <input
        type="text"
        :value="value"
        @input="$emit('input', $event.target.value)"
        @keyup.enter="$emit('keyupEnter')"
        id="add-option-input"
      >
    </b-col>
  </b-row>
</template>

<script>
export default {
  name: 'PollOptioninput',
  props: ['value']
}
</script>

<style>
#add-option-input {
  width: 100%;
}
</style>
