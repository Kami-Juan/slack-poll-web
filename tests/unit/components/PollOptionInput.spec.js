import BootstrapVue from 'bootstrap-vue'
import { LayoutPlugin, ModalPlugin } from 'bootstrap-vue'

import { mount, createLocalVue } from '@vue/test-utils'
import PollOptionInput from './../../../src/components/Poll/PollOptionInput'

describe('test PollOptionInput.vue', () => {
  const localVue = createLocalVue()

  localVue.use(BootstrapVue)
  localVue.use(LayoutPlugin)
  localVue.use(ModalPlugin)

  it('should be render', () => {
    const cmp = mount(PollOptionInput, {localVue})
    expect(cmp.element).toMatchSnapshot()
  })

  it('the poll input has the correct value', () => {
    const cmp = mount(PollOptionInput, {localVue, propsData: {
      value: 'I"m poll'
    }})

    let pollInput = cmp.find('#add-option-input').element.value
    expect(pollInput).toEqual('I"m poll')
  })

  it('set data in the poll input and fire the input event', () => {
    const cmp = mount(PollOptionInput, {localVue})

    const pollInput = cmp.find('#add-option-input')
    pollInput.setValue('I"m poll two')

    expect(cmp.emitted().input).toBeTruthy()
    expect(cmp.emitted().input.length).toBe(1)
    expect(cmp.emitted().input[0]).toEqual(['I"m poll two'])
  })

  it('set attribute in the component and the poll input get it', () => {
    const cmp = mount(PollOptionInput, {localVue, attrs: {
      type: 'password',
      name: 'input-password',
      placeholder: 'Add an amazing option'
    }})

    const pollInput = cmp.find('#add-option-input').attributes()

    expect(pollInput.type).toBe('password')
    expect(pollInput.name).toBe('input-password')
    expect(pollInput.placeholder).toBe('Add an amazing option')
  })

  it('set listener and in the component and the poll input fire it', async () => {
    const clickHandler = jest.fn()

    const cmp = mount(PollOptionInput, {localVue, listeners: {
      keyup: clickHandler
    }})

    const pollInput = cmp.find('#add-option-input')
    pollInput.trigger('keyup.enter')

    expect(clickHandler).toHaveBeenCalled()
  })
})