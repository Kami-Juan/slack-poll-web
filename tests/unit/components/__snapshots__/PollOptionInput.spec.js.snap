// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PollOptionInput.vue should be render 1`] = `
<div
  class="row justify-content-center"
  id="poll-option-input"
>
  <div
    class="col-md-6 col-12"
  >
    <input
      id="add-option-input"
    />
  </div>
</div>
`;
